File: Java Programming/Competitive Programming/Coding Challenges 11/program4.java
//Q4.Write a program which accept number from user and return summation of all its non factors.
//Input : 12
//Output : 50
//Input : 10
//Output : 37

import java.util.Scanner;

class program4
{
    static int SumNonFact(int iNo)
    {
        if (iNo < 0)
        {
            iNo = -iNo;
        }

        int i = 1;
        int iTotal = 0;
        int iFactSum = 0;

        while (i <= iNo)
        {
            iTotal = iTotal + i;

            if (iNo % i == 0)
            {
                iFactSum = iFactSum + i;
            }

            i++;
        }

        return iTotal - iFactSum;
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        int iRet = 0;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = SumNonFact(iValue);

        System.out.print("Output is : " + iRet);

        sobj.close();
    }
}

File: C++ Programming/Competitive Programming/Coding Challenges 13/program4.cpp
/*
4.Write a program which accept number from user and return multiplication of all
digits.
Input : 2395
Output : 270
Input : 1018
Output : 8
Input : 9440
Output : 144
Input : 922432
Output : 864
*/

#include <iostream>
using namespace std;

int MultDigits(int iNo)
{
    int iMult = 1;

    if (iNo < 0)
    {
        iNo = -iNo;
    }

    for (; iNo != 0; iNo = iNo / 10)
    {
        int iDigit = iNo % 10;

        if (iDigit > 0)
        {
            iMult = iMult * iDigit;
        }
    }

    return iMult;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout << "Enter number: ";
    cin >> iValue;

    iRet = MultDigits(iValue);

    cout << "Output is : " << iRet;

    return 0;
}